### PR TITLE
Lexer parser

### PR DESCRIPTION
**Pull request by kilowatt-**

### CODE REVIEW

File: src/parser/parser.ts [hunk 1]
```diff
@@ -1,8 +1,242 @@
...
+    const args: Id[] = [];
+
+    this.expect(keyword, TokenType.LITERAL, "import");
+
+    const identifierToken = this.lexer.lexRegular();
+    this.lexer.unlex(identifierToken);
+
+    if (identifierToken.type === TokenType.ID) {
+      id = this.parseId();
+      this.lexer.unlex(identifierToken);
```
Comment (JustinHuPrime) [inline src/parser/parser.ts:58]: You're unlexing this twice

File: src/parser/parser.ts [hunk 2]
```diff
@@ -1,8 +1,242 @@
...
+    }
+
+    const openParan = this.lexer.lexRegular();
+
+    this.expect(openParan, TokenType.LITERAL, "(");
+
+    const arg1 = this.parseId();
+    args.push(arg1);
```
Comment (JustinHuPrime) [inline src/parser/parser.ts:66]: We aren't guarenteed to have an id, are we?

File: src/parser/parser.ts [hunk 3]
```diff
@@ -1,8 +1,242 @@
...
+          this.lexer.unlex(current);
+          const id = this.parseId();
+          args.push(id);
+          break;
+        case TokenType.LITERAL:
+          this.expect(current, TokenType.LITERAL, ",");
+          break;
+        default:
+          throw new ParserError(`${this.filename}: ${line}:${character}: unexpected token '${content}`);
+          break;
```
Comment (JustinHuPrime) [inline src/parser/parser.ts:86]: This probably needs to be refactored - you allow multiple commas in a row

File: src/parser/parser.ts [hunk 4]
```diff
@@ -1,8 +1,242 @@
...
+  private parsePrint(): Expression {
+    const keyword = this.lexer.lexRegular();
+    this.expect(keyword, TokenType.LITERAL, "print");
+
+    const expression = this.parseExpression();
+
+    return new Print(keyword, expression);
+  }
+
+  // @ts-ignore
```
Comment (JustinHuPrime) [inline src/parser/parser.ts:183]: Why's this needed?

Comment (kilowatt-) [reply src/parser/parser.ts:183]: Oh, this is temporary, I'll take it out later - basically TSC is unhappy with the code as it is now

File: src/parser/parser.ts (hunk 3 above)
Comment (kilowatt-) [reply src/parser/parser.ts:86]: Right, thanks. I thought about that then completely forgot about it while coding it up.

File: src/parser/parser.ts (hunk 2 above)
Comment (kilowatt-) [reply src/parser/parser.ts:66]: Oh yeah, good point. I read the grammar wrongly

**Pull request merged**